File: _python/1.py
# -*- coding: utf-8 -*-
# @Time    : 2018/1/20 下午8:58
# @Author  : wxy
# @File    : 1.py

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()
        for i in xrange(len(nums)):
            if d.get(target-nums[i],None)!=None:
                return [d[target-nums[i]],i]
            d[nums[i]]=i




if __name__ == '__main__':
    l = [2, 7,11,15]
    t = 9
    print Solution().twoSum(l, t)

File: _cpp/answer/1.cpp

#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

namespace p1 {
    class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            unordered_map<int,int> m;
            vector<int> res;
            for (int i = 0; i < nums.size(); ++i) {
                if (m.find(nums[i])!=m.end()){
                    res.push_back(m[nums[i]]);
                    res.push_back(i);
                    return res;
                }
                m[target-nums[i]]=i;
            }
        }
    };
}